baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/ConsistentOverridesTest.java

```diff
@@ -0,0 +1,167 @@
+                        "  interface Foo {",
+                        "    void doStuff(String foo, int bar);",
+                        "  }",
+                        "  class DefaultFoo implements Foo {",
+                        "    @Override",
+                        "    public void doStuff(String bar, int foo) {}",
+                        "  }",
+                        "}")
+                .expectUnchanged()
+                .doTest();
+    }
+
+    @Test
+    void ignores_unambiguous_generics_rename() {
+        fix().addInputLines(
+                        "Test.java",
+                        "import " + List.class.getCanonicalName() + ";",
+                        "class Test {",
+                        "  interface Foo {",
+                        "    void doStuff(List<String> foo, List<Integer> bar);",
+                        "  }",
+                        "  class DefaultFoo implements Foo {",
+                        "    @Override",
+                        "    public void doStuff(List<String> bar, List<Integer> foo) {}",
+                        "  }",
+                        "}")
+                .expectUnchanged()
+                .doTest();
+    }
+
+    @Test
+    void swapped_names() {
+        fix().addInputLines(
+                        "Test.java",
+                        "class Test {",
+                        "  interface Foo {",
+                        "    boolean doStuff(String foo, String bar);",
+                        "  }",
+                        "  class DefaultFoo implements Foo {",
+                        "    @Override",
+                        "    public boolean doStuff(String bar, String foo) {",
+                        "      return bar.equals(foo);",
+                        "    }",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "class Test {",
+                        "  interface Foo {",
+                        "    boolean doStuff(String foo, String bar);",
+                        "  }",
+                        "  class DefaultFoo implements Foo {",
+                        "    @Override",
+                        "    public boolean doStuff(String foo, String bar) {",
+                        "      return foo.equals(bar);",
+                        "    }",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void overridden_name() {
+        fix().addInputLines(
+                        "Test.java",
+                        "class Test {",
+                        "  interface Foo {",
+                        "    void doStuff(String foo, String bar);",
+                        "  }",
+                        "  class DefaultFoo implements Foo {",
+                        "    @Override",
+                        "    public void doStuff(String foo, String baz) {}",
```

Comment:
Can we add a pair of tests for underscore prefixed names? For example `void doStuff(String foo, String bar)` should be implementable as `@Override public void doStuff(String foo, String _bar)` if `bar` isn't used, while the implementation `@Override public void doStuff(String foo, String _baz)` should be updated to `@Override public void doStuff(String foo, String _bar)`

Reply:
I added a test to demonstrate that it plays nicely with `StrictUnusedVariable`. I avoided actually making `_baz` be changed to `_bar` since it effectively would be duplicating the logic of `StrictUnusedVariable`. Perhaps we say that if a variable is unused then it doesn't really matter what its name is?

Reply:
Sounds good to me, in that case I suppose we still enforce that the other variable matches the supertype? Things might get weird if we override a class that defines a method with an unused arg, but we want to use it?

Reply:
yup we still enforce that the first parameter is `foo` regardless of what the second one is. Ya thats a bit funky. I've made it so that if the super param is `_bar` you can name the param `_bar` or `bar`

Reply:
:+1: